Wrappers/Python/ccpi/viewer/CILViewer.py

```diff
@@ -170,105 +171,95 @@ def OnRightMouseRelease(self, interactor, event):
         if not (alt and ctrl and shift):
             self.SetEventInactive("ZOOM_EVENT")
 
-    def OnKeyPress(self, interactor, event):
+    def ToggleSliceInterpolation(self):
+        # toggle interpolation of image slice
+        is_interpolated = self._viewer.imageSlice.GetProperty().GetInterpolationType()
+        if is_interpolated:
+            self._viewer.imageSlice.GetProperty().SetInterpolationTypeToNearest()
+        else:
+            self._viewer.imageSlice.GetProperty().SetInterpolationTypeToLinear()
+        self._viewer.updatePipeline()
 
-        ctrl = interactor.GetControlKey()
-        shift = interactor.GetAltKey()
-        alt = interactor.GetShiftKey()
+    def ToggleVolumeClipping(self):
+        viewer = self._viewer
+        viewer.imageSlice.VisibilityOff()
+        # clip a volume render if available
+        if hasattr(self._viewer, 'planew') and self._viewer.clipping_plane_initialised:
+            is_enabled = viewer.planew.GetEnabled()
+            viewer.planew.SetEnabled(not is_enabled)
+            viewer.getRenderer().Render()
+        else:
+            planew = self.CreateClippingPlane()
+            planew.On()
+        viewer.updatePipeline()
+
+    def ToggleSliceVisibility(self):
+        # toggle visibility of the slice
+        if self._viewer.imageSlice.GetVisibility():
+            self._viewer.imageSlice.VisibilityOff()
+        else:
+            self._viewer.imageSlice.VisibilityOn()
+        self._viewer.updatePipeline()
+
+    def ToggleVolumeVisibility(self):
+        # toggle visibility of the volume render
+        if not self._viewer.volume_render_initialised:
+            self._viewer.installVolumeRenderActorPipeline()
+
+        if self._viewer.volume.GetVisibility():
+            self._viewer.volume.VisibilityOff()
+            self._viewer.light.SwitchOff()
+        else:
+            self._viewer.volume.VisibilityOn()
+            self._viewer.light.SwitchOn()
+        self._viewer.updatePipeline()
 
+    def ResetVolume(self):
```

Comment:
This should have a more specific name such as ResetVolumeWindowLevel

Reply:
or AutoVolumeWindowLevel

Reply:
Done :+1: